spec/spec_helper.rb

```diff
@@ -3,12 +3,14 @@
   SimpleCov.start
 end
 
-Dir[Rails.root.join("spec/shared/**/*.rb")].each { |f| require f }
-Dir[File.join(__dir__, "support/**/*.rb")].each { |f| require f }
+Dir[Rails.root.join("spec/shared/**/*.rb")].sort.each { |f| require f }
+Dir[File.join(__dir__, "support/**/*.rb")].sort.each { |f| require f }
 
 require "manageiq-providers-oracle_cloud"
 
 VCR.configure do |config|
   config.ignore_hosts 'codeclimate.com' if ENV['CI']
   config.cassette_library_dir = File.join(ManageIQ::Providers::OracleCloud::Engine.root, 'spec/vcr_cassettes')
+  config.define_cassette_placeholder("ocid1.user.oc1..aaaaaaaa") { Rails.application.secrets.oracle_cloud[:user_id] }
+  config.define_cassette_placeholder("ocid1.tenancy.oc1..aaaaaaaa") { Rails.application.secrets.oracle_cloud[:tenant_id] }
```

Comment:
Interesting...I wonder if there's a way we can access this key...something like...

In secrets.defaults.yml
```yaml
---
test:
  oracle_cloud_defaults: &oracle_cloud_defaults
    user_id: ocid1.user.oc1..aaaaaaaa
    ...
  oracle_cloud:
     <<: *oracle_cloud_defaults
```

Then here

```ruby
  config.define_cassette_placeholder(Rails.application.secrets.oracle_cloud_defaults[:user_id]) do
    Rails.application.secrets.oracle_cloud[:user_id]
  end
```

Reply:
This way overrides will be in the oracle_cloud section, but the defaults would still be accessible in the oracle_cloud_defaults section.

Reply:
For example locally I have this:

```ruby
[3] pry(main)> require "yaml"
[4] pry(main)> x = YAML.load("test:
  oracle_cloud_defaults: &oracle_cloud_defaults
    user_id: ocid1.user.oc1..aaaaaaaa
  oracle_cloud: *oracle_cloud_defaults")
=> {"test"=>{"oracle_cloud_defaults"=>{"user_id"=>"ocid1.user.oc1..aaaaaaaa"}, "oracle_cloud"=>{"user_id"=>"ocid1.user.oc1..aaaaaaaa"}}}
[5] pry(main)> x["test"]["oracle_cloud"]["user_id"] = "foo"
=> "foo"
[6] pry(main)> x
=> {"test"=>{"oracle_cloud_defaults"=>{"user_id"=>"ocid1.user.oc1..aaaaaaaa"}, "oracle_cloud"=>{"user_id"=>"foo"}}}
```

Reply:
Yeah I like the defaults, don't like a second top-level group but what about
```
oracle_cloud:
  defaults: &oracle_cloud_defaults
    user_id: xxx
    tenant_id: xxx
  <<: *oracle_cloud_defaults
```

This way it is all under `Rails.application.secrets.oracle_cloud`

Reply:
oh nice!

Reply:
:( unfortunately when I have a `config/secrets.yml` it doesn't merge but rather overrides everything under `test.oracle_cloud` so the `defaults` get blown away